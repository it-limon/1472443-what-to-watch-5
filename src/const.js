export const MAX_COMMENTS_PER_COLUMN = 2;
export const MAX_GENRES_COUNT = 9;
export const SIMILAR_MOVIES_COUNT = 4;
export const DEFAULT_SHOWN_MOVIES_COUNT = 8;
export const DEFAULT_REVIEW_RATING = 3;
export const MIN_COMMENT_LENGTH = 50;
export const MAX_COMMENT_LENGTH = 400;
export const ALL_GENRES = `All genres`;

export const MovieRating = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very Good`,
  AWESOME: `Awesome`
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

export const AppPages = {
  MAIN: `MAIN`,
  AUTH: `AUTH`,
  MOVIE: `MOVIE`,
  MYLIST: `MYLIST`,
  REVIEW: `REVIEW`,
  PLAYER: `PLAYER`
};

export const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  FILMS: `/films`,
  MYLIST: `/mylist`,
  REVIEW: `/review`,
  PLAYER: `/player`
};

export const APIRoute = {
  FILMS: `/films`,
  FAVORITE_FILMS: `/favorite`,
  PROMO_MOVIE: `/films/promo`,
  COMMENTS: `/comments`,
  LOGIN: `/login`
};
