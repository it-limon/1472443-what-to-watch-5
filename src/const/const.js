export const MovieRatings = [`Bad`, `Normal`, `Good`, `Very Good`, `Awesome`];
export const REVIEW_COLUMNS_COUNT = 2;
export const LIKE_THIS_MOVIE_COUNT = 4;
