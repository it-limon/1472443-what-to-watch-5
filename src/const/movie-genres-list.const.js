export const MovieGenresList = [
  {
    key: 0,
    ord: 0,
    name: `All genres`,
    filterName: `All genres`,
  },
  {
    key: 1,
    ord: 1,
    name: `Comedie`,
    filterName: `Comedies`,
  },
  {
    key: 2,
    ord: 2,
    name: `Crime`,
    filterName: `Crime`,
  },
  {
    key: 3,
    ord: 3,
    name: `Documentary`,
    filterName: `Documentary`,
  },
  {
    key: 4,
    ord: 4,
    name: `Drama`,
    filterName: `Dramas`,
  },
  {
    key: 5,
    ord: 5,
    name: `Horror`,
    filterName: `Horror`,
  },
  {
    key: 6,
    ord: 6,
    name: `Kids & Family`,
    filterName: `Kids & Family`,
  },
  {
    key: 7,
    ord: 7,
    name: `Romance`,
    filterName: `Romance`,
  },
  {
    key: 8,
    ord: 8,
    name: `Sci-Fi`,
    filterName: `Sci-Fi`,
  },
  {
    key: 9,
    ord: 9,
    name: `Thriller`,
    filterName: `Thrillers`,
  }
];
