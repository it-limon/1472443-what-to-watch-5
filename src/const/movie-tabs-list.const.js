export const MovieTabsList = [
  {
    key: 0,
    name: `Overview`
  },
  {
    key: 1,
    name: `Details`
  },
  {
    key: 2,
    name: `Reviews`
  }
];
