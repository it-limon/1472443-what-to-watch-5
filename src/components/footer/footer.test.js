import React from "react";
import renderer from "react-test-renderer";
import Footer from "./footer";
import {Route, BrowserRouter} from "react-router-dom";
import {AppPages} from "../../const";

describe(`Render Footer`, () => {

  it(`Render Footer with active link`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <Route>
              <Footer
                currentPage={AppPages.MOVIE}
              />
            </Route>
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render Footer without active link`, () => {
    const tree = renderer
      .create(
          <Footer
            currentPage={AppPages.MAIN}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
