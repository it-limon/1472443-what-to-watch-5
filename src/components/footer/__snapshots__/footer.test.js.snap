// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Footer Render Footer with active link 1`] = `
<footer>
  <a
    className="logo__link logo__link--light"
    href="/"
    onClick={[Function]}
  >
    <span
      className="logo__letter logo__letter--1"
    >
      W
    </span>
    <span
      className="logo__letter logo__letter--2"
    >
      T
    </span>
    <span
      className="logo__letter logo__letter--3"
    >
      W
    </span>
  </a>
  <div
    className="copyright"
  >
    <p>
      © 2019 What to watch Ltd.
    </p>
  </div>
</footer>
`;

exports[`Render Footer Render Footer without active link 1`] = `
<footer>
  <div
    className="logo__link logo__link--light"
  >
    <span
      className="logo__letter logo__letter--1"
    >
      W
    </span>
    <span
      className="logo__letter logo__letter--2"
    >
      T
    </span>
    <span
      className="logo__letter logo__letter--3"
    >
      W
    </span>
  </div>
  <div
    className="copyright"
  >
    <p>
      © 2019 What to watch Ltd.
    </p>
  </div>
</footer>
`;
